src/handlers/psp.ts

```diff
@@ -0,0 +1,436 @@
+import * as abi from '@api3/airnode-abi';
+import * as node from '@api3/airnode-node';
+import { ethers } from 'ethers';
+import { Dictionary } from 'lodash';
+import flatMap from 'lodash/flatMap';
+import groupBy from 'lodash/groupBy';
+import isEmpty from 'lodash/isEmpty';
+import isNil from 'lodash/isNil';
+import map from 'lodash/map';
+import { callApi } from '../api/call-api';
+import { loadAirnodeConfig, mergeConfigs, parseConfig } from '../config';
+import { checkSubscriptionCondition } from '../evm/check-conditions';
+import { initializeProvider } from '../evm/initialize-provider';
+import { processTransaction } from '../evm/process-transaction';
+import { getSponsorWalletAndTransactionCount } from '../evm/tansaction-count';
+import {
+  ChainConfig,
+  CheckedSubscription,
+  Config,
+  EVMProviderState,
+  GroupedSubscriptions,
+  Id,
+  ProviderState,
+  SponsorWalletTransactionCount,
+  SponsorWalletWithSubscriptions,
+  State,
+  Subscription,
+} from '../types';
+import { retryGo } from '../utils';
+
+export const handler = async (_event: any = {}): Promise<any> => {
+  const startedAt = new Date();
+
+  const airnodeConfig = loadAirnodeConfig();
+  // This file will be merged with config.json from above
+  const airkeeperConfig: Config = parseConfig('airkeeper');
+  const config = mergeConfigs(airnodeConfig, airkeeperConfig);
+
+  const state = await updateBeacon(config);
+
+  const completedAt = new Date();
+  const durationMs = Math.abs(completedAt.getTime() - startedAt.getTime());
+  node.logger.info(
+    `PSP beacon update finished at ${node.utils.formatDateTime(completedAt)}. Total time: ${durationMs}ms`,
+    state.baseLogOptions
+  );
+
+  const response = {
+    ok: true,
+    data: { message: 'PSP beacon update execution has finished' },
+  };
+  return { statusCode: 200, body: JSON.stringify(response) };
+};
+
+const initializeState = (config: Config): State => {
+  const { triggers, subscriptions } = config;
+
+  const baseLogOptions = node.logger.buildBaseOptions(config, {
+    coordinatorId: node.utils.randomHexString(8),
+  });
+
+  const enabledSubscriptions: Id<Subscription>[] = [];
+  triggers['proto-psp'].forEach((subscriptionId) => {
```

Comment:
This can be turned into a `map`, I kind of did it already:
```
  const enabledSubscriptions = config.triggers['proto-psp']
    .filter((subscriptionId) => config.subscriptions[subscriptionId])
    .map((subscriptionId) => ({
      ...config.subscriptions[subscriptionId],
      id: subscriptionId,
    }))
    .filter((subscription) => {
      const subscriptionId = subscription.id;
      // Get subscriptions details
      if (isNil(subscription)) {
        node.logger.warn(`SubscriptionId ${subscriptionId} not found in subscriptions`, baseLogOptions);
        return;
      }
      // Verify subscriptionId
      const expectedSubscriptionId = ethers.utils.solidityKeccak256(
        ['uint256', 'address', 'bytes32', 'bytes', 'bytes', 'address', 'address', 'address', 'bytes4'],
        [
          subscription.chainId,
          subscription.airnodeAddress,
          subscription.templateId,
          subscription.parameters,
          subscription.conditions,
          subscription.relayer,
          subscription.sponsor,
          subscription.requester,
          subscription.fulfillFunctionId,
        ]
      );
      if (subscriptionId !== expectedSubscriptionId) {
        node.logger.warn(
          `SubscriptionId ${subscriptionId} does not match expected ${expectedSubscriptionId}`,
          baseLogOptions
        );
        return;
      }

      return true;
    });

  if (isEmpty(enabledSubscriptions)) {
    node.logger.info('No proto-psp subscriptions to process', baseLogOptions);

    return {
      config,
      baseLogOptions,
      groupedSubscriptions: [],
      apiValuesBySubscriptionId: {},
      providerStates: [],
    };
  }
```

Reply:
Yes, I was expecting to refactor this into a promise.all(). If you have the fix and have already tested it please feel free to push it. Otherwise I'll do since I know how busy you are these days.

Reply:
I'm not so busy now, but I tested most of these in my IDE before commenting them :smile: 
Sometime my brain won't leave things alone :/